tests/PatternAssertions/BasePatternAssert.php

```diff
@@ -113,7 +113,11 @@ protected function assertElementText($expected, string $selector, Crawler $crawl
     }
     $this->assertElementExists($selector, $crawler);
     $element = $crawler->filter($selector);
-    self::assertEquals($expected, trim($element->text()));
+    $actual = trim($element->text());
+    self::assertEquals($expected, $actual, \sprintf(
```

Comment:
```suggestion
    $this->assertEquals($expected, $actual, \sprintf(
```
Can we use $this-> instead of each self:: occurance?

Reply:
I think `self` is used intentionally. Might be it is design to allow subclasses to override methods like assertCount. But as we are in the base class we want to use original assertCount.
@imanoleguskiza what was your thoughts when you have implemented this?

Reply:
I think here its the opposite, we are using self so we force the usage of the original methods defined in the phpunit Assert class.

Reply:
yes, I mean the same.